ext/include/opentelemetry/ext/http/common/url_parser.h

```diff
@@ -0,0 +1,127 @@
+#pragma once
+
+#include <string>
+#include <vector>
+#include "opentelemetry/nostd/string_view.h"
+#include "opentelemetry/version.h"
+OPENTELEMETRY_BEGIN_NAMESPACE
+namespace ext
+{
+namespace http
+{
+namespace common
+{
+// http://user:password@host:port/path1/path2?key1=val2&key2=val2
+// http://host:port/path1/path2?key1=val1&key2=val2
+// host:port/path1
+// host:port ( path defaults to "/")
+// host:port?
+
+class UrlParser
```

Comment:
Could `curl_get_url` do the work instead of creating our own parser?

Reply:
The idea is to have a cross platform url parser usable throughout sdk/exporters. As an example here : https://github.com/open-telemetry/opentelemetry-cpp/blob/c93fe62ef651ac2d2018161338e8d33d4afdb38d/ext/test/w3c_tracecontext_test/main.cc#L55